### ISSUE TITLE
Starting with data episode generates a few issues in R 4.0.0

### ISSUE THREAD

**Issue by cjlortie** [datacarpentry/R-ecology-lesson#609]
R version 4.0.0 (2020-04-24)
Platform: x86_64-apple-darwin17.0 (64-bit)
Running under: macOS Catalina 10.15.4
Starting with data episode
1. In the first challenge, 'Based on the output of str(surveys), can you answer the following questions?', there are 40 species not 48 as proposed in the answer.

2. In the Renaming factors subsection, plot(surveys$sex)returns an error and warning messages because read.csv read in the vector sex as character not factor in R 4.0.0. Subsequent code such as levels(sex) etc return NULL so this solution would be a good learning revision 
sex <- factor(surveys$sex)

**Comment by justinshaffer** [comment]
Hello!

Regarding (1), it seems that with categorical data read in as characters vs. factors, the answer to the third question in the first challenge cannot be discerned as the output from str(surveys) does not provide the number of levels for surveys$species. This Challenge could be re-tooled to accommodate this question, or alternatively differences between character vs. factor could be explored elsewhere as you suggest in (2).

Thoughts?

**Comment by jebyrnes** [comment]
Hrm....

```
surveys$species_factor <- as.factor(surveys$species)
str(surveys)

....

$ species_factor : Factor w/ 40 levels "albigula","audubonii",..: 1 1 1 1 1 1 1 1 1 1 ...
```


Also

```
> length(unique(surveys$species))
[1] 40
```

I think the issue is, there are 48 if you combine genus and species

```
> length(unique(paste(surveys$genus, surveys$species)))
[1] 48
```

**Comment by jebyrnes** [comment]
OK, I'm an idiot, and I apologies - the `species_id` column has 48 unique entries - perhaps we can clarify which column they need to look at. HA!

**Comment by Teebusch** [comment]
1 - Indeed, the species column has multiple entries that are "sp.". For non-ecologists, the hierarchy of genus and species may not be obvious. I think this issue can be solved easily by adding a short note to the challenge (e.g., "hint: use the `species_id` column" to find all unique species)

2 - The issue with stringsAsFactors in R>4.0 is larger. One temporary and not too confusing workaround would be to first show `table(surveys$sex)` and then `barplot(table(surveys$sex))`. This independent of whether the column is a factor or a string vector and it that sense it might be most "future proof".

**Comment by Teebusch** [comment]
Related to Issue #471

**Comment by fmichonneau** [comment]
addressed with #663
